### PR TITLE
core iptables settings

### PR DESCRIPTION
**Pull request by hans0r**
please review iptables changes and validate

### CODE REVIEW

**Comment by blenessy** [comment]
can you please incorporate your (very useful) changes to `basic-node.yml` instead of `basic-node-hans0r.yml` ?

File: examples/docker-compose/basic-node-hans0r.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        ipv4_address: 172.16.88.2
+        aliases:
+         - webserver
+volumes:
+  core_ledger:
+  core_config:
+  nginx_secrets:
+
+networks:
+  radix_net:
```
Comment (blenessy) [inline examples/docker-compose/basic-node-hans0r.yml]: Why is this necessary ? Isn't it better to allow `docker-compose` to create a node-specific (bridged) network automatically ?

File: examples/docker/radixdlt-core/docker-entrypoint.sh [hunk 2]
```diff
@@ -37,4 +37,9 @@ if [ $max_mb -gt 4096 ]; then
     export JAVA_OPTS="-Xmx$(($max_mb - 2560))m $JAVA_OPTS"
 fi
 
+# load iptables
+if [ -f /sbin/iptables-restore ]; then
```
Comment (blenessy) [inline examples/docker/radixdlt-core/docker-entrypoint.sh]: lets do this uncoditionally (assume `/sbin/iptables-restore` is available). `docker run` will fail otherwise, which is exactly what we want.

File: examples/docker/radixdlt-core/docker-entrypoint.sh [hunk 3]
```diff
@@ -37,4 +37,9 @@ if [ $max_mb -gt 4096 ]; then
...
+iptables-restore < /etc/iptables_restore/iptables.rules
```
Comment (blenessy) [inline examples/docker/radixdlt-core/docker-entrypoint.sh]: Please use absolute path to `/sbin/iptables-restore` so that it cannot be overloaded in PATH.

File: examples/docker/radixdlt-core/iptables.rules [hunk 4]
```diff
@@ -0,0 +1,21 @@
+*filter
+:INPUT ACCEPT [503:30180]
```
Comment (blenessy) [inline examples/docker/radixdlt-core/iptables.rules]: Where does 503:30180 come from ? Why is it better then 0:0 ?

File: examples/docker/radixdlt-core/iptables.rules [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+:FORWARD ACCEPT [0:0]
+:OUTPUT ACCEPT [1265:75900]
```
Comment (blenessy) [inline examples/docker/radixdlt-core/iptables.rules]: Where does 1265:75900 come from and why is better than 0:0 ?

File: examples/docker/radixdlt-core/iptables.rules [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+# if there are packages dropped we can log this in two ways jump LOG/ULOG/NFLOG must be specified before log-prefix
+# every time
+# -A RATE-LIMIT -j LOG --log-prefix "IPTables-Rejected: "
+# only if there are more then 5 per second
+# -A RATE-LIMIT -m limit --limit 5/sec -j LOG --log-prefix "IPTables-Rejected: "
+# silently drop or
+-A RATE-LIMIT -j DROP
+#reject with icmp unreachable
+# -A RATE-LIMIT -j REJECT --reject-with icmp-port-unreachable
+COMMIT
```
Comment (blenessy) [inline examples/docker/radixdlt-core/iptables.rules]: Please add a trailing newline otherwise `iptables-restore` will refuse this.

File: examples/docker/radixdlt-core/iptables.rules [hunk 7]
```diff
@@ -0,0 +1,21 @@
+*filter
+:INPUT ACCEPT [503:30180]
+:FORWARD ACCEPT [0:0]
+:OUTPUT ACCEPT [1265:75900]
+:RATE-LIMIT - [0:0]
+#:DOCKER - [0:0]
+# define a explicit rule for 20000 udp/tcp because it is possible that there are much more requests on the http-api
+-A INPUT -m conntrack -p tcp --dport 20000 --ctstate NEW -j RATE-LIMIT
+-A INPUT -m conntrack -p udp --dport 20000 --ctstate NEW -j RATE-LIMIT
+-A RATE-LIMIT -m hashlimit --hashlimit-upto 50/sec --hashlimit-burst 20 --hashlimit-name DOS --hashlimit-htable-size 16348 --hashlimit-htable-max 16348 --hashlimit-htable-gcinterval 3000 --hashlimit-htable-expire 300000 -j ACCEPT
```
Comment (blenessy) [inline examples/docker/radixdlt-core/iptables.rules]: please inc `--hashlimit-upto` 500/sec and  `--hashlimit-burst` 500 .

File: examples/docker/radixdlt-core/Dockerfile.alpine [hunk 8]
```diff
@@ -2,14 +2,17 @@ ARG ARCH=amd64
...
-RUN apk --no-cache --update add openjdk8 libc6-compat
+RUN apk --no-cache --update add openjdk8 libc6-compat iptables
 
 COPY docker-healthcheck.sh /
 HEALTHCHECK CMD sh /docker-healthcheck.sh
 
 COPY docker-entrypoint.sh /
 ENTRYPOINT ["/bin/sh", "/docker-entrypoint.sh"]
 
+RUN mkdir /etc/iptables_restore
```
Comment (blenessy) [inline examples/docker/radixdlt-core/Dockerfile.alpine]: not necessary - please remove

File: examples/docker/radixdlt-core/Dockerfile.alpine [hunk 9]
```diff
@@ -2,14 +2,17 @@ ARG ARCH=amd64
...
+COPY iptables.rules /etc/iptables_restore
```
Comment (blenessy) [inline examples/docker/radixdlt-core/Dockerfile.alpine]: Why not just `/etc/iptables` ?

File: examples/docker/radixdlt-core/iptables.rules (hunk 5 above)
Comment (hans0r) [reply examples/docker/radixdlt-core/iptables.rules]: It is not better.. it was a copy & paste fault! I've corrected that. That is the network package count.

File: examples/docker/radixdlt-core/iptables.rules (hunk 4 above)
Comment (hans0r) [reply examples/docker/radixdlt-core/iptables.rules]: It is not better.. it was a copy & paste fault! I've corrected that. That is the network package count.

File: examples/docker-compose/basic-node-hans0r.yml (hunk 1 above)
Comment (hans0r) [reply examples/docker-compose/basic-node-hans0r.yml]: This comes from some of my tests because I needed a fixed IP inside the container. I can delete this and update the normal basic-node.yml

File: examples/docker/radixdlt-core/docker-entrypoint.sh (hunk 2 above)
Comment (hans0r) [reply examples/docker/radixdlt-core/docker-entrypoint.sh]: done.

File: examples/docker/radixdlt-core/iptables.rules (hunk 6 above)
Comment (hans0r) [reply examples/docker/radixdlt-core/iptables.rules]: hopefully fixed.. wrong version on server/mac

File: examples/docker/radixdlt-core/iptables.rules (hunk 7 above)
Comment (hans0r) [reply examples/docker/radixdlt-core/iptables.rules]: done

File: examples/docker/radixdlt-core/docker-entrypoint.sh (hunk 3 above)
Comment (hans0r) [reply examples/docker/radixdlt-core/docker-entrypoint.sh]: fixed

File: examples/docker/radixdlt-core/Dockerfile.alpine (hunk 8 above)
Comment (hans0r) [reply examples/docker/radixdlt-core/Dockerfile.alpine]: done

File: examples/docker/radixdlt-core/Dockerfile.alpine (hunk 9 above)
Comment (hans0r) [reply examples/docker/radixdlt-core/Dockerfile.alpine]: fixed with next push. We can copy it to - that has no specific background

**Comment by hans0r** [comment]
please review changes.. entrypoint with su-exec must be added from your side :-) @blenessy

**Comment by blenessy** [comment]
LGTM, thanks @hans0r .
